bundles/org.openhab.binding.easee/src/main/java/org/openhab/binding/easee/internal/handler/EaseeChargerHandler.java

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    /**
+     * Disposes the thing.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        cancelJobReference(dataPollingJobReference);
+    }
+
+    /**
+     * will update all channels provided in the map
+     */
+    @Override
+    public void updateChannelStatus(Map<Channel, State> values) {
+        logger.debug("Handling charger channel update.");
+
+        for (Channel channel : values.keySet()) {
+            if (getThing().getChannels().contains(channel)) {
+                State value = values.get(channel);
+                if (value != null) {
+                    logger.debug("Channel is to be updated: {}: {}", channel.getUID().getAsString(), value);
+                    updateState(channel.getUID(), value);
+                } else {
+                    logger.debug("Value is null or not provided by Easee Cloud (channel: {})",
+                            channel.getUID().getAsString());
+                    updateState(channel.getUID(), UnDefType.UNDEF);
+                }
+            } else {
+                logger.debug("Could not identify channel: {} for model {}", channel.getUID().getAsString(),
+                        getThing().getThingTypeUID().getAsString());
+            }
+        }
+    }
+
+    @Override
+    public void enqueueCommand(EaseeCommand command) {
+        EaseeBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            bridgeHandler.enqueueCommand(command);
+        } else {
+            // this should not happen
+            logger.warn("no bridge handler found");
+        }
+    }
+
+    private @Nullable EaseeBridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+        return bridge == null ? null : ((EaseeBridgeHandler) bridge.getHandler());
+    }
+
+    @Override
+    public EaseeConfiguration getBridgeConfiguration() {
+        EaseeBridgeHandler bridgeHandler = getBridgeHandler();
+        return bridgeHandler == null ? new EaseeConfiguration() : bridgeHandler.getBridgeConfiguration();
+    }
+
+    @Override
+    public EaseeCommand buildEaseeCommand(Command command, Channel channel) {
+        String chargerId = getConfig().get(EaseeBindingConstants.THING_CONFIG_ID).toString();
+
+        switch (Utils.getWriteCommand(channel)) {
+            case COMMAND_CHANGE_CONFIGURATION:
+                return new ChangeConfiguration(this, chargerId, channel, command);
+            case COMMAND_SEND_COMMAND:
+                return new SendCommand(this, chargerId, channel, command);
+            case COMMAND_SEND_COMMAND_START_STOP:
+                return new SendCommandStartStop(this, chargerId, channel, command);
+            default:
+                // this should not happen
+                logger.warn("write command '{}' not found for channel '{}'", command.toString(),
```

Comment:
Probably the binding will stop working after this since exception is not caught, so this could even be logged as error.

Reply:
someone told me that "error" is reserved for framework erros while "warn" can be used for binding errors. This is why I used "warn" only. But I agree this is a severe error which should not occur. So I will change it to level "error"